Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MATLABAnalytics/MATLABAnalytic.tsx

```diff
@@ -0,0 +1,107 @@
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  08/07/2023 - Parker Dinsdale
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+
+
+import * as React from 'react';
+import MATLABAnalyticInfo from './MATLABAnalyticInfo';
+import MATLABAnalyticSQLSettings from './MATLABAnalyticSQLSetting';
+import { useAppSelector, useAppDispatch } from '../hooks';
+import { MATLABAnalyticSlice, MATLABAnalyticEventTypeSlice, MATLABAnalyticAssetTypeSlice } from '../Store/Store';
+import { TabSelector, Warning } from '@gpa-gemstone/react-interactive';
+
+declare var homePath: string;
+
+export default function MATLABAnalytic(props: { AnalyticID: number }) {
+    const dispatch = useAppDispatch();
+
+    const record = useAppSelector((state) => MATLABAnalyticSlice.Datum(state, props.AnalyticID));
+    const eventTypeRecords = useAppSelector(MATLABAnalyticEventTypeSlice.Data);
+    const assetTypeRecords = useAppSelector(MATLABAnalyticAssetTypeSlice.Data);
+
+    const status = useAppSelector(MATLABAnalyticSlice.Status);
+    const eventTypeStatus = useAppSelector(MATLABAnalyticEventTypeSlice.Status);
+    const assetTypeStatus = useAppSelector(MATLABAnalyticAssetTypeSlice.Status);
+
+    const [tab, setTab] = React.useState<'info' | 'settings'>('info');
+    const [showRemove, setShowRemove] = React.useState<boolean>(false);
+
+    React.useEffect(() => {
+        if (status == 'unintiated' || status == 'changed')
+            dispatch(MATLABAnalyticSlice.Fetch());
+    }, [status]);
+
+    React.useEffect(() => {
+        if (eventTypeStatus == 'unintiated' || eventTypeStatus == 'changed')
+            dispatch(MATLABAnalyticEventTypeSlice.Fetch(props.AnalyticID));
+    }, [eventTypeStatus]);
+
+    React.useEffect(() => {
+        if (assetTypeStatus == 'unintiated' || assetTypeStatus == 'changed')
+            dispatch(MATLABAnalyticAssetTypeSlice.Fetch(props.AnalyticID));
+    }, [assetTypeStatus]);
+
+    React.useEffect(() => {
+        sessionStorage.setItem('MATLABAnalytic.Tab', JSON.stringify(tab));
+    }, [tab]);
+
+    function Delete() {
+        dispatch(MATLABAnalyticSlice.DBAction({ verb: 'DELETE', record }));
+        eventTypeRecords.forEach((item) => dispatch(MATLABAnalyticEventTypeSlice.DBAction({ verb: 'DELETE', record: item })));
+        assetTypeRecords.forEach((item) => dispatch(MATLABAnalyticAssetTypeSlice.DBAction({ verb: 'DELETE', record: item })));
```

Comment:
```suggestion
```
@clackner-gpa correct me if I am wrong, but this should be unnecessary, since this records should get caught and deleted by the cascading delete on SQL side.

Reply:
I tested deleting a record without these lines and yes it seems that they are caught in the cascading delete. I have removed them.